semantics/cpp14/language/translation/name.k

```diff
@@ -497,6 +497,76 @@ module CPP-TRANSLATION-NAME
           => resolveEmptyOverload(cSetUnion2(C1, C2), list(ListItem(E1) ListItem(E2)), E1 -> no-template E2, BinaryOperator(operator->, E1, E2))
 
+     // 13.6:15
+/*
+     rule addBuiltinCandidates(operator<, E1::Expr, T1:CPPEnumType, _, E2::Expr, T2:CPPEnumType, _, C1:CandidateSet, C2:CandidateSet)
+          => resolveEmptyOverload(cSetUnion2(C1, C2), list(ListItem(E1) ListItem(E2)), E1 < E2, BinaryOperator(operator<, E1, E2))
+          requires simpleType(T1) ==K simpleType(T2)
+
+     rule addBuiltinCandidates(operator>, E1::Expr, T1:CPPEnumType, _, E2::Expr, T2:CPPEnumType, _, C1:CandidateSet, C2:CandidateSet)
+          => resolveEmptyOverload(cSetUnion2(C1, C2), list(ListItem(E1) ListItem(E2)), E1 > E2, BinaryOperator(operator>, E1, E2))
+          requires simpleType(T1) ==K simpleType(T2)
+
+     rule addBuiltinCandidates(operator<=, E1::Expr, T1:CPPEnumType, _, E2::Expr, T2:CPPEnumType, _, C1:CandidateSet, C2:CandidateSet)
+          => resolveEmptyOverload(cSetUnion2(C1, C2), list(ListItem(E1) ListItem(E2)), E1 <= E2, BinaryOperator(operator<=, E1, E2))
+          requires simpleType(T1) ==K simpleType(T2)
+
+     rule addBuiltinCandidates(operator>=, E1::Expr, T1:CPPEnumType, _, E2::Expr, T2:CPPEnumType, _, C1:CandidateSet, C2:CandidateSet)
+          => resolveEmptyOverload(cSetUnion2(C1, C2), list(ListItem(E1) ListItem(E2)), E1 >= E2, BinaryOperator(operator>=, E1, E2))
+          requires simpleType(T1) ==K simpleType(T2)
+
+     rule addBuiltinCandidates(operator==, E1::Expr, T1:CPPEnumType, _, E2::Expr, T2:CPPEnumType, _, C1:CandidateSet, C2:CandidateSet)
+          => resolveEmptyOverload(cSetUnion2(C1, C2), list(ListItem(E1) ListItem(E2)), E1 == E2, BinaryOperator(operator==, E1, E2))
+          requires simpleType(T1) ==K simpleType(T2)
+
+     rule addBuiltinCandidates(operator!=, E1::Expr, T1:CPPEnumType, _, E2::Expr, T2:CPPEnumType, _, C1:CandidateSet, C2:CandidateSet)
+          => resolveEmptyOverload(cSetUnion2(C1, C2), list(ListItem(E1) ListItem(E2)), E1 != E2, BinaryOperator(operator!=, E1, E2))
+          requires simpleType(T1) ==K simpleType(T2)
+*/
+
+
+     syntax Map ::= builtinNonvolatileEnumAssignOperator(Expr, CPPType, Expr, CPPType, ValueCategory) [function]
+     syntax Map ::= builtinVolatileEnumAssignOperator(Expr, CPPType, Expr, CPPType, ValueCategory) [function]
+     syntax Map ::= builtinNonvolatileArithAssignOperator(Expr, CPPType, ValueCategory, Expr, CPPType, ValueCategory) [function]
+
+     rule builtinNonvolatileEnumAssignOperator(E1::Expr, T1::CPPType, E2::Expr, T2::CPPType, VC::ValueCategory) =>
+                (type(functionType(type(simpleType(T1)),
+                      toCPPTypes(ListItem(type(lvRefType(type(simpleType(T1))))) ListItem(type(simpleType(T1)))),
+                      noMethod, CPPLinkage, exSet(.Set)))
+                |-> builtinOp(E1 := E2))
+
+          requires isCPPEnumType(T1) andBool notBool isVolatileType(T1) andBool viableType(type(simpleType(T1)), T2, VC)
```

Comment:
I am not sure with the non-volatility requirement. I would think we should keep it, because non-volatile argument is not convertible to volatile type (and 13.3.1.2:3.3.3).